tools/PI/DevHome.PI/Controls/AddToolControl.xaml

```diff
@@ -17,89 +19,123 @@
             <RowDefinition Height="Auto"/>
         </Grid.RowDefinitions>
 
-        <StackPanel Orientation="Vertical">
+        <Grid>
+            <Grid.RowDefinitions>
```

Comment:
Seems like the rows declare a top-margin of 6px.  You can declare the spacing here instead.

Reply:
Not all rows have the same margins.